thrust/testing/regression/gh_1706__thrust_optional_reference_emplace.cu

```diff
@@ -0,0 +1,22 @@
+#include <thrust/optional.h>
+#include <unittest/unittest.h>
+
+int main()
+{
+    int a = 10;
```

Comment:
Out of pure paranoia, please add a test that ensures we can emplace an empty optional